client/pages/view-skills-by-role.tsx

```diff
@@ -0,0 +1,89 @@
+import React from "react";
+import { useState } from 'react';
+import { Card, Divider, Grid, Page, Spacer, Text } from '@geist-ui/core';
+import { Skill, skillCategories } from '../database/skills';
+
+const SkillFind: React.FC = () => {
+    const skills= [
+        {
+            role : 'Sales',
+            label : 'Product Knowledge',
+            description : 'Description is empty',
+        },
+        {
+            role : 'Technology',
+            label : 'Web Development',
+            description : 'Description is empty',
+        }
+    ];
+    const [skillList, setSkillList] = React.useState<
+        { role:string; label:string; description:string; }[] | undefined
+    >(skills);
+    const [text, setText] = React.useState<string>('');
+
+    const handleOnClick = () => {
+        const findSkills = 
+            skillList && skillList?.length > 0 
+            ? skillList?.filter((s) => s?.role == text)
+            : undefined;
+
+            console.log(findSkills);
+        setSkillList(findSkills);
+    };
+
+    const [selectedSkills, setSelectedSkills] = useState<string[]>([])
+    return(
+        <div>
+            <div className="title">
+            <h1>View Skills by Role</h1>
+            </div>
+            <div className="input_wrapper">
+                <input 
+                type="text" 
+                placeholder="Search Skills" 
+                value={text} 
+                onChange={(e) => {
+                    setText(e.target.value); 
+                    setSkillList(skills);
+                }}
+                />
+                <button disabled={!text} onClick={handleOnClick}>Search</button>
+
+            </div>
+                <div className="body">
+
+                    {skillList && skillList?.length == 0 &&(
+                        <div className="notFound">No Skills Found</div>
+                    )}
+                {skillList && 
+                    skillList?.length>0 && 
+                    skillList?.map(role =>{
+                    return(
+                        <div className="body__item">
```

Comment:
```suggestion
                        <div className="body__item" key={role}>
```
This should fix the error :D (you can view the error by running `tsc --noEmit` under the client folder). Every list-rendered component would need a key prop for React to identify it for maintaining its sequence. (ref: [this](https://reactjs.org/docs/lists-and-keys.html)).

Reply:
Here are the error logs that show why the build is failing:

`62:25  Error: Missing "key" prop for element in iterator  react/jsx-key`
![image](https://user-images.githubusercontent.com/62209188/196310252-c15ad790-d5a0-4c01-9273-7b764be649d5.png)

Reply:
If I add `key={role}` , I'll have error on the code
![image](https://user-images.githubusercontent.com/89002933/196350302-2bd9ea74-9ff9-436a-b31c-a36d0d4b0315.png)

Reply:
What's the error shown when you hover on `key`?

Reply:
There is no error in terminal and website works fine. Its just that there the `key` is underlined. I just push it.